include/heat_transfer/heat_transfer.h

```diff
@@ -685,12 +689,54 @@ namespace Heat_Transfer
   }
 
 
-
   template <int dim>
   void
   LaplaceProblem<dim>::solve()
   {
-    TimerOutput::Scope               t(timer, "solve system");
+    TimerOutput::Scope t(timer, "solve system");
+    SolverControl      solver_control(100, 1e-12 * system_rhs.l2_norm());
+
+    std::string preconditioner_type = "gmg";
+    if (preconditioner_type == "jacobi")
+      {
+        // FIXME: Leads currently to more iterations compared to "none"
+        SolverCG<VectorType>                 cg(solver_control);
+        PreconditionJacobi<SystemMatrixType> preconditioner;
+        double                               relaxation = .7;
+        preconditioner.initialize(system_matrix, relaxation);
+        system_matrix.compute_diagonal();
+        cg.solve(system_matrix, solution_update, system_rhs, preconditioner);
+      }
```

Comment:
Alternatively, you can use `PreconditionRelaxation`. You can attach a diagonal preconditioner there to get a Jacobi preconditioner (the usage is the same as `PreconditionChebyshev`). Regarding the relaxation parameter, I tend to use the eigenvalue estimates from `PreconditionChebyshev` like this:

```cpp
        const auto degree = params.get<unsigned int>("degree", 3);
        auto       omega  = params.get<double>("omega", 0.0);




        pcout << "- Create system preconditioner: Relaxation" << std::endl;
        pcout << "    - degree: " << degree << std::endl;

        if (omega == 0.0)
          {
            const auto chebyshev =
              create_chebyshev_preconditioner(op, precon, params);


            VectorType vec;
            op.initialize_dof_vector(vec);
            const auto evs = chebyshev->estimate_eigenvalues(vec);


            const unsigned int smoothing_range = 20;


            const double alpha =
              (smoothing_range > 1. ?
                 evs.max_eigenvalue_estimate / smoothing_range :
                 std::min(0.9 * evs.max_eigenvalue_estimate,
                          evs.min_eigenvalue_estimate));

            omega = 2.0 / (alpha + evs.max_eigenvalue_estimate);


            pcout << "    - min ev: " << evs.min_eigenvalue_estimate
                  << std::endl;
            pcout << "    - max ev: " << evs.max_eigenvalue_estimate
                  << std::endl;
            pcout << std::endl;
          }

        pcout << "    - omega:  " << omega << std::endl;

        typename RelaxationPreconditionerType::AdditionalData additional_data;
        additional_data.preconditioner = precon;
        additional_data.n_iterations   = degree;
        additional_data.relaxation     = omega;
```

Reply:
> Alternatively, you can use PreconditionRelaxation. You can attach a diagonal preconditioner there to get a Jacobi preconditioner (the usage is the same as PreconditionChebyshev).

What's the point of replacing the jacobian preconditioner by `PreconditionRelaxation`, if it results in the same operator? Adjusting the the number of iterations, however, improved the preconditioner performance significantly. I also tried the eigenvalue estimate, but the relaxation parameter seems to have only a limited influence for `n_iterations=1`. Overall, the jacobi preconditioner performance for `n_iterations=1` is still worse compared to `Identity` which seems a bit counter intuitive. I will integrate the changes later.